extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -6,7 +6,7 @@ const https = require('https');
 const { EventEmitter } = require('events');
 const Redis = require('ioredis');
 const schedule = require('node-schedule');
-const promClient = require('prom-client');
+const { ZenkoMetrics } = require('arsenal').metrics;
```

Comment:
What is the reason behind going to zenko metrics from the prom client?

Reply:
Initial reason was consistency, since ZenkoMetrics is used everywhere. It also automatically inject the "default" node metrics.

Then i found out that not using it is an issue: other parts of the code use it, and actually end up using _another copy_ of prom-client via arsenal, and thus another singleton registry. Some metrics end up in ZenkoMetrics/arsenal copy (backbeatconsumer, backbeatproducer, default metrics) and replication ones go in directly imported copy: and it is ugly to merge both in the route callback.

Not sure why ZenkoMetrics was introduced in the first place, but I just followed through to avoid issues and collect all metrics as expected.

Reply:
The default node metrics were intentionally not imported as they were very bloaty and did not seem to be helpful given the size increase. Backbeat consumer and producer not including there metrics could be a problem, I am not sure this effects 7x so maybe we just ignored it. I believe we were also concerned about trying to limit what metrics were being registered given backbeat has many processes in one project. I am not sure if this actually matters though.

Oh and by wrapping the prom client we lose all type hints which makes working with the metrics a little harder. Not a big deal as we wrap the metrics with the metric handler but that part is at least mostly type aware when using the prom client directly. 

Personally I think a static wrapper around a static registry is a bit odd and problematic, hence why we did not use zenko metrics and was trying to deprecate the use over time ( similar to the health check route and the probe server update ).

Reply:
Are there any measure as to the bloat of default metrics ?
Not having them was actually an issue on KB, to help us isolate issue related to node itself...

As for ZenkoMetrics itself, I fully agree we should not need a static wrapper: yet I don't think I have the time to remove it overall (and validate the change), so I'll prefer to stay consistent for now, and leave it for another day.

Reply:
We did not measure them I don't think. We had them on at first and when I presented them there were some 40 or so metrics I think, compared to the like 3 we added. When I mentioned this to Rahul we chose to turn the defaults off to reduce prometheus memory usage.